trino/client.py

```diff
@@ -839,3 +768,147 @@ def decorated(*args, **kwargs):
         return decorated
 
     return wrapper
+
+
+class RowMapperFactory:
+    """
+    Given the 'columns' result from Trino, generate a list of
+    lambda functions (one for each column) which will process a data value
+    and returns a RowMapper instance which will process rows of data
+    """
+
+    def create(self, columns, experimental_python_types):
+        assert columns is not None
+
+        if experimental_python_types:
+            return RowMapper([self._col_func(column['typeSignature']) for column in columns])
+        return RowMapper()
+
+    def _col_func(self, column):
+        col_type = column['rawType']
```

Comment:
What other fields does the typeSignature have available? I'm asking because it's possible for startswith to fail in presence of user contributed types.

Reply:
typeSignature has `rawType` and `arguments` fields (the latter is used for types such as `decimal(3, 1)`)

Reply:
Then can the checks below use `rawType == 'decimal'` instead of existing `rawType.startswith('decimal')` etc.?

Reply:
I've seen cases where the value is `decimal(18, 18)`

Reply:
👀 Thanks. I'll need to re-read some code on the server side to have better understanding. 🙂